src/sql/src/plan/transform_ast.rs

```diff
@@ -613,6 +613,39 @@ impl<'a> Desugarer<'a> {
             }
         }
 
+        // When `$expr` is a `ROW` constructor, we need to desugar as described
+        // below in order to enable the row comparision expansion at the end of
+        // this function. We don't do this desugaring unconditionally (i.e.,
+        // when `$expr` is not a `ROW` constructor) because the implementation
+        // in `plan_in_list` is more efficient when row comparison expansion is
+        // not required.
+        //
+        // `$expr IN ($list)` => `$expr = $list[0] OR $expr = $list[1] ... OR
+        // $expr = $list[n]` `$expr IN ($list)` => `$expr <> $list[0] AND $expr
```

Comment:
If I understand this correctly,

```suggestion
        // $expr = $list[n]` `$expr NOT IN ($list)` => `$expr <> $list[0] AND $expr
```

Reply:
Ugh, thanks, fixed.